### DISCUSSION TITLE
Network host on windows

### DISCUSSION THREAD

**Discussion by Enelg52** [ThePorgs/Exegol#87] [Q&A] [answered]
Hi ! 
I just tried Exegol on a Windows 10 machine and I have a question. My container settings are saying that I'm using the host network. But the docker documentation says that it does only work on a Linux host. https://docs.docker.com/network/network-tutorial-host/
Is this a mistake ? 
![image](https://user-images.githubusercontent.com/70370923/166158298-2f48e400-8f1e-428a-bec7-e21b3aa8ef8f.png)

**Comment by Dramelac** [comment]
Indeed you are right, this particularity of docker desktop is not taken in consideration by exegol which still tries to create the container in host mode.
Could you do a docker inspect of your container to see the network configuration of the container?

**Reply by Enelg52** [reply]
This is what I got :
```
"NetworkSettings": {
            "Bridge": "",
            "SandboxID": "b1d0e8c0db75a2a40215e3cedc6c1efeb224bae992165864485f71ac00852514",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/var/run/docker/netns/default",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "host": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "91654775c1f1e63d4ccf04098e5112d075a13fb9be47c9b77bba7ff945ae761d",
                    "EndpointID": "eb413357485f9ed10b334b562c8557adb21d37a7b3a2d10f12a4ace0e3653bf8",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
```

**Comment by Dramelac** [comment, marked as answer]
The subject will be clarified in the next release of Exegol (4.0.2).
It will **no longer be possible** to create containers in host mode from this version however it will now be possible to **open ports** (docker NAT) between the host and the container with the `-p` or `--port` parameter.
The port opening does not cover all the problems, however there is always the possibility to **open VPN** connections

**Reply by Dramelac** [reply]
Now available on master with #92
